src/mod/l4_thruster.c

```diff
@@ -792,114 +461,79 @@ void GeneralSetRequest(int argc, char *argv[]){
 		// at all registers with only 1 byte data length conversion multiplier array
 		// will also be an integer value. Therefore typecast of conversion multiplier
 		//into uint8_t should also work
 		input_uint8 = input_uint8* (uint8_t) CONVERSION_DOUBLE[access_register];
-
 		request[7]= input_uint8;
 
 	}
 
-	/////////////////////////////////////////here in this block error
 	if (length_of_register ==2){
 
 		// parse argument as double
-		double input = atof((const char*)argv[2]); // WARNING :::: ATOF DOES NOT WORK
+		// WARNING : Set Project-Settings-Manager linker script - Redlib (nohost) to use atof()
+		double input = atof((const char*)argv[2]);
 		// in cases where data stored in register has length of 2 bytes. uint16_t would be used
 		// to store this data in register. For some registers input may be float
 		// example 3.3V  or 3.0 or 3
 		// Function should be able to handle both float and integer inputs from user
 
-
-		//APPLY CONVERSION MULTIPLIER
+		// Apply conversion multiplier
 		input = input* CONVERSION_DOUBLE[access_register];
-		// CONVERT INPUT INTO UINT16
+		// Convert input to uint16_t
 		uint16_t value = (uint16_t) input;
-
-
-		// represent uint16_t as two uint8_t bytes to fill the request array.
+		// Represent uint16_t as two uint8_t bytes to fill the request array.
 		request[7]= value & 0xff;
 		request[8] = (value >> 8) & 0xff;
-
 	}
-	//////////////////////
-
-
 
 	request[3] = CRC8(request,len); // calculate checksum after whole request array is sent
 	l4_thr_ExpectedReceiveBuffer = 6;// change expected receive buffer accordingly
-
 	thrSendBytes(request, len);
-    //printf("%s",request);
-
-
-
-
 }
 
 
-/////// GENERAL READ REQUEST TO ANY REGISTER
+//General read request to any register
 void GeneralReadRequest(int argc, char *argv[]){
 
 		// FIRST ARGUMENT SHOUD BE int VALUE OF REGISTER THAT WOULD BE READ FROM
 		uint8_t access_register = atoi(argv[1]);
-
 		uint8_t length_of_register = REGISTER_LENGTH[access_register];
 
-		//Check if valid/existing register is accessed
-
+		// TODO : Check if valid/existing register is accessed
 		if(length_of_register ==0){
-			return; //as previously discussed - 0 entry in length array means no info
-			//therefore not proceed with that request
+			return;
 		}
 
-
 		uint8_t request[8];
 		request[0] = SENDER_ADRESS;
 		request[1] = DEVICE;
 		request[2] = MSGTYPE[2]; // READ -3
-		request[3] = 0x00; //checksumm
+		request[3] = 0x00; // checksum
 		request[4] = 0x02; // LENGTH of payload REGISTER and length
-		request[5] = 0x00; //hardcoded because length of payload is defined with two bytes of uint16
-		request[6]= REGISTER_VALUES[access_register]; // Access register at address requested by used
-
-        //printf("\n register value is set to hex %02X  int %d\n",request[6],request[6]);
+		request[5] = 0x00; // Hardcoded because length of payload is defined with two bytes of uint16
+		request[6]= access_register; // Access register at address requested by used
```

Comment:
no need to use REGISTER_VALUES array. 

long time ago when I initialized it - I thought that requster addresses are not sequential